From 0353dd999aa4964cd313393d51c76a3b00962c1e Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Tue, 18 Mar 2025 09:54:05 +0100
Subject: [PATCH 1617/1695] dts: stm32: fix: bad GPU clock on device tree

Kernel clock should be used to set frenquencies of OPPs.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fda0a67401db932bb0a640795afd04cf679124cf dts: stm32: fix: bad GPU clock on device tree]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I816dd732ccd9f6e55f8e63d2f2c4085d88ff2c76
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/444788
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp235.dtsi | 2 +-
 core/arch/arm/dts/stm32mp255.dtsi | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp235.dtsi b/core/arch/arm/dts/stm32mp235.dtsi
index 7c4dd7ace..7fb7c2cf8 100644
--- a/core/arch/arm/dts/stm32mp235.dtsi
+++ b/core/arch/arm/dts/stm32mp235.dtsi
@@ -23,7 +23,7 @@
 	gpu: gpu@48280000 {
 		compatible = "vivante,gc";
 		gpu-supply = <&vddgpu>;
-		clocks = <&rcc CK_BUS_GPU>;
+		clocks = <&rcc CK_KER_GPU>;
 		operating-points-v2 = <&gpu_opp_table>;
 		access-controllers = <&rifsc STM32MP25_RIFSC_GPU_ID>;
 	};
diff --git a/core/arch/arm/dts/stm32mp255.dtsi b/core/arch/arm/dts/stm32mp255.dtsi
index 1c221ead6..9212171d3 100644
--- a/core/arch/arm/dts/stm32mp255.dtsi
+++ b/core/arch/arm/dts/stm32mp255.dtsi
@@ -29,7 +29,7 @@
 	gpu: gpu@48280000 {
 		compatible = "vivante,gc";
 		gpu-supply = <&vddgpu>;
-		clocks = <&rcc CK_BUS_GPU>;
+		clocks = <&rcc CK_KER_GPU>;
 		operating-points-v2 = <&gpu_opp_table>;
 		access-controllers = <&rifsc STM32MP25_RIFSC_GPU_ID>;
 	};
-- 
2.47.2

